packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js

```diff
@@ -0,0 +1,201 @@
+import { TRANSLATE } from './accesses.constants';
+
+export default class NashaDashboardPartitionAccessesController {
+  /* @ngInject */
+  constructor($q, $http, $translate, iceberg) {
+    this.$q = $q;
+    this.$http = $http;
+    this.$translate = $translate;
+    this.iceberg = iceberg;
+
+    this.isAccessFormShown = false;
+    this.isCreatingAccess = false;
+    this.isLoadingAccessForm = false;
+    this.isLoadingAccesses = false;
+    this.accesses = [];
+    this.authorizedAccesses = null;
+    this.model = { ip: null, type: null };
+    this.typeOptions = [];
+  }
+
+  $onInit() {
+    this.typeOptions = this.aclTypeEnum
+      .filter((type) =>
+        type === 'readonly' ? this.partition.protocol === 'NFS' : true,
+      )
+      .map((type) => ({
+        label: this.translate(`list_type_${type}`),
+        value: type,
+      }));
+  }
+
+  get datagridTypeOptions() {
+    return {
+      hideOperators: true,
+      values: this.typeOptions.reduce(
+        (result, { label, value }) => ({
+          ...result,
+          [value]: label,
+        }),
+        {},
+      ),
+    };
+  }
+
+  get canCreateAccess() {
```

Comment:
Are getters useful in an angularjs controller ? I'd understand the need in a TS/JS class but here I wonder if it is really needed

Reply:
An angularjs controller **is a class**
IMO this is perfectly valid because it avoids using a method, it is more readable, it makes much more sense if the returned value is a primitive (in this case, a boolean). The digest cycle works normally. You can see it as if it were a vuejs computed property

Please take a look at https://blog.thoughtspile.tech/2018/09/20/angularjs-service-property-getter/

Another place where I used it in combination with a setter https://github.com/ovh/manager/pull/6864/files#diff-3f3d97fed0e3c4c727b23f1f6423078c9f45b65cd3ae88c0adf1ae54ad259206R43-R49